### PR TITLE
Bugs-Not yet processed submissions are not protected from another retest

### PR DESCRIPTION
**Pull request by gogo4ds**
Judge system doubles tests https://github.com/SoftUni-Internal/suls-issues-public/issues/69

### CODE REVIEW

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs [hunk 1]
```diff
@@ -582,6 +582,13 @@ private void RetestSubmission(DatabaseModelType submission)
                 return;
             }
 
+            var submissionForProcessing = this.Data.SubmissionsForProcessing.GetById(submission.Id);
+
+            if (submissionForProcessing != null && !submissionForProcessing.Processed)
```
Comment (jicata) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs]: Is this check sufficient to determine whether the _actual_ submission has been `Processed` ? What if it  isn't present in the SubmissionForProcessing table?

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs [hunk 2]
```diff
@@ -582,6 +582,13 @@ private void RetestSubmission(DatabaseModelType submission)
                 return;
             }
 
+            var submissionForProcessing = this.Data.SubmissionsForProcessing.GetById(submission.Id);
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs]: Shouldn't  you use another method to get the submission like GetBySubmissionId, unless GetById is overriden it uses a default implementation that gets a submission by it's primary key Id (which for the SubmissionsForProcessing table is not the Submission id from what I understand).

File: Open Judge System/Web/OJS.Web/App_GlobalResources/Areas/Administration/Submissions/SubmissionsControllers.resx [hunk 3]
```diff
@@ -132,6 +132,9 @@
   <data name="Submission_content_length_invalid" xml:space="preserve">
     <value>Submission is over the size limit!</value>
   </data>
+  <data name="Submission_is_processing" xml:space="preserve">
+    <value>Submission is processig and cannot be retested!</value>
```
Comment (jicata) [inline Open Judge System/Web/OJS.Web/App_GlobalResources/Areas/Administration/Submissions/SubmissionsControllers.resx]: "Submission is being processed and cannot be retested right now"

File: Open Judge System/Web/OJS.Web/App_GlobalResources/Areas/Administration/Submissions/SubmissionsControllers.bg.resx [hunk 4]
```diff
@@ -132,6 +132,9 @@
   <data name="Submission_content_length_invalid" xml:space="preserve">
     <value>Решението надвишава лимита за големина!</value>
   </data>
+  <data name="Submission_is_processing" xml:space="preserve">
+    <value>Решението се обработва и не може да бъде ретествано!</value>
```
Comment (jicata) [inline Open Judge System/Web/OJS.Web/App_GlobalResources/Areas/Administration/Submissions/SubmissionsControllers.bg.resx]: "Решението се обработва в момента и не може да бъде ретествано!"

**Pull request merged**